sources/engine/Stride.Engine/Profiling/DebugTextSystem.cs

```diff
@@ -49,13 +50,36 @@ public void Print(string message, Int2 position)
         /// <param name="color"></param>
         public void Print(string message, Int2 position, Color4 color)
         {
-            var msg = new DebugOverlayMessage { Message = message, Position = position, TextColor = color };
-            overlayMessages.Enqueue(msg);
+            Print(message, position, color, DefaultOnScreenTime);
+        }
+
+        /// <summary>
+        /// Print a custom overlay message
+        /// </summary>
+        /// <param name="message"></param>
+        /// <param name="position"></param>
+        /// <param name="timeOnScreen"></param>
+        public void Print(string message, Int2 position, TimeSpan timeOnScreen)
+        {
+            Print(message, position, TextColor, timeOnScreen);
+        }
+
+        /// <summary>
+        /// Print a custom overlay message
+        /// </summary>
+        /// <param name="message"></param>
+        /// <param name="position"></param>
+        /// <param name="color"></param>
+        /// <param name="timeOnScreen"></param>
+        public void Print(string message, Int2 position, Color4 color, TimeSpan timeOnScreen)
+        {
+            var msg = new DebugOverlayMessage { Message = message, Position = position, TextColor = color, RemeaningTime = timeOnScreen };
+            overlayMessages.Add(msg);
 
             //drop one old message if the tail size has been reached
             if (overlayMessages.Count > TailSize)
             {
-                overlayMessages.Dequeue();
+                overlayMessages.RemoveAt(0);
             }
         }
```

Comment:
The amount of signatures is getting a bit out of hand, how about we remove all of them and add this one in instead:
```cs
public void Print(string message, Int2 position, Color4? color = null, TimeSpan? timeOnScreen = null)
{
    var msg = new DebugOverlayMessage { Message = message, Position = position, TextColor = color ?? TextColor, RemeaningTime = timeOnScreen ?? DefaultOnScreenTime };
    overlayMessages.Add(msg);
...
```
With that one you could call it in the following ways:
```cs
Print("msg", pos, timeOnScreen: TimeSpan.FromSeconds(1));
Print("msg", pos, Color.LightGreen, TimeSpan.FromSeconds(1));
Print("msg", pos, null, TimeSpan.FromSeconds(1));
Print("msg", pos);
```

Reply:
On the beggining i tried to follow what was already there but doing it that way you can do the same things with less code.
Also you can also use named parameters and just ommit the ones that you don't care about too
```cs
Print(message: "test", position: new Int2(), color: null, timeOnScreen: null);
Print(message: "test", position: new Int2(), timeOnScreen: null);
Print(message: "test", position: new Int2(), color: null);
Print(message: "test", position: new Int2());
```